docs/03-infrastructure-deployment.md

```diff
@@ -0,0 +1,304 @@
+- **SKU**: PerGB2018 (pay-as-you-go)
+- **Data retention**: 30 days
+
+### 4. Container Apps Environment
+
+The hosting environment for Container Apps:
+- Provides networking isolation
+- Integrates with Log Analytics
+- Manages shared configuration
+
+### 5. User-Assigned Managed Identity
+
+Provides identity for the container app to access Azure resources:
+- Eliminates need for storing secrets
+- Granted appropriate RBAC roles on OpenAI and AI Search
+- Used for pulling images from ACR
+
+### 6. Azure OpenAI Service
+
+Provides GPT and embedding models:
+- **SKU**: S0 (Standard)
+- **Deployments**:
+  - GPT model: `gpt-4.1-mini` (chat completions)
+  - Embedding model: `text-embedding-3-small` (document embeddings)
+- **Capacity**: 50 tokens per minute (configurable)
+- **Authentication**: Both API key and passwordless (managed identity)
+
+#### Role Assignments:
+- Cognitive Services OpenAI User (for user principal)
+- Cognitive Services OpenAI Contributor (for user principal)
+- Cognitive Services OpenAI User (for container app managed identity)
+
+### 7. Azure AI Search Service
+
+Vector database for document storage and retrieval:
+- **SKU**: Basic (supports semantic search)
+- **Semantic search**: Standard tier
+- **Partition count**: 1
+- **Replica count**: 1
+- **Index name**: `northwind` (configured in environment variables)
+- **Schema**: Uses default schema for vector documents (compatible with LangChain)
+
+#### Role Assignments:
+- Search Index Data Contributor (for creating and updating index)
+- Search Index Data Reader (for querying)
+- Search Service Contributor (for managing service)
+
+### 8. Azure Container App
+
+The running application:
+- **Container**: Built from `Dockerfile` in repository root
+- **Initial image**: Placeholder (updated during `azd deploy`)
+- **Resources**: 0.5 CPU, 1Gi memory
+- **Ingress**: External HTTPS with port 3000
+- **Scaling**: 1-3 replicas
+- **Registry**: Connected to ACR using managed identity
+
+#### Environment Variables:
+All configuration is injected as environment variables:
+- Azure AI Search endpoint and index name
+- Azure OpenAI instance names and model names
+- API versions for completions and embeddings
+- `SET_PASSWORDLESS=true` for managed identity authentication
+- `AZURE_CLIENT_ID` for the managed identity
+
+## Deployment Commands
+
+### Full Deployment
+
+```bash
+# Deploy everything (infrastructure + application)
+azd up
+```
+
+This runs the complete deployment pipeline:
+1. Validates configuration
+2. Provisions infrastructure (or updates if exists)
+3. Builds Docker image
+4. Pushes image to ACR
+5. Deploys to Container Apps
+6. Loads vector data
+
+### Incremental Commands
+
+```bash
+# Provision/update infrastructure only
+azd provision
+
+# Deploy application code only (infrastructure must exist)
+azd deploy
+
+# View environment variables
+azd env get-values
+
+# Set environment variable
+azd env set KEY value
+
+# Delete all resources
```

Comment:
add --purge and note that removes the resources and any frees quota limitation the resources may have had. For example, if you remove individual resources in the Azure portal, that quota is help for 48 hours, or until the resource is purged.